Library/Formula/ibex.rb

```diff
@@ -0,0 +1,38 @@
+class Ibex < Formula
+  homepage "http://www.ibex-lib.org/"
+  url "http://www.ibex-lib.org/sites/default/files/ibex-2.1.12.tgz"
+  version "2.1.12"
+  sha1 "9ba6f72309a732cc6347e1f6f30c7b6581ff20c2"
+
+  option "with-java", "Build Java bindings for Choco solver."
+  if build.with? "java"
+      depends_on :java => "1.8+"
+      depends_on 'pkg-config' => :build
+  end
+  depends_on 'gcc' => :build
```

Comment:
I added the dependency on gcc only because g++ is needed at build. I searched for an example that uses fails_with for this but I can't find any.